src/core/mac/mac.cpp

```diff
@@ -870,6 +882,8 @@ extern "C" void otPlatRadioTransmitDone(otInstance *aInstance, otRadioFrame *aFr
 {
     otLogFuncEntryMsg("%!otError!, aRxPending=%u", aError, aRxPending ? 1 : 0);
 
+    aInstance->mThreadNetif.GetMac().SetFramePending(aRxPending);
```

Comment:
Suggestion: Instead of checking the "frame pending" at the tx done callback, we can consider checking it when we issue the send command (from  within `RadioTrasnmit()`). Good thing about this is that we can then do the check inside the `#if OPENTHREAD_CONFIG_STAY_AWAKE_BETWEEN_FRAGMENTS` block (I think if the feature is disbaled, we don't need to  keep track of "pending next tx frames").